core/src/test/java/kafka/admin/AclCommandTest.java

```diff
@@ -0,0 +1,729 @@
+                        ADD, "--resource-pattern-type", patternType.toString()));
+                verifyPatternType(addCmd, patternType.isSpecific());
+
+                List<String> listCmd = new ArrayList<>(cmdArgs);
+                listCmd.addAll(Arrays.asList("--topic", "Test", LIST, "--resource-pattern-type", patternType.toString()));
+                verifyPatternType(listCmd, patternType != PatternType.UNKNOWN);
+
+                List<String> removeCmd = new ArrayList<>(cmdArgs);
+                removeCmd.addAll(Arrays.asList("--topic", "Test", "--force", REMOVE, "--resource-pattern-type", patternType.toString()));
+                verifyPatternType(removeCmd, patternType != PatternType.UNKNOWN);
+            });
+        } finally {
+            Exit.resetExitProcedure();
+        }
+    }
+
+    private void verifyPatternType(List<String> cmd, boolean isValid) {
+        if (isValid)
+            callMain(cmd);
+        else
+            assertThrows(RuntimeException.class, () -> callMain(cmd));
+    }
+
+    private void testRemove(
+            ClusterInstance cluster,
+            List<String> cmdArgs,
+            Set<ResourcePattern> resources,
+            List<String> resourceCmd
+    ) {
+        List<String> args = new ArrayList<>(cmdArgs);
+        args.addAll(resourceCmd);
+        args.add(REMOVE);
+        args.add("--force");
+        Map.Entry<String, String> out = callMain(args);
+        assertEquals("", out.getValue());
+        for (ResourcePattern resource : resources) {
+            withAuthorizer(cluster, authorizer ->
+                    TestUtils.waitAndVerifyAcls(asScalaSet(Collections.emptySet()), authorizer, resource, ANY));
+        }
+    }
+
+    private Map.Entry<Set<AccessControlEntry>, List<String>> getAclToCommand(
+            AclPermissionType permissionType,
+            Set<AclOperation> operations
+    ) {
+        return new SimpleImmutableEntry<>(
+                asJavaSet(AclCommand.getAcls(asScalaSet(USERS), permissionType, asScalaSet(operations), asScalaSet(HOSTS))),
+                getCmd(permissionType)
+        );
+    }
+
+    private List<String> getCmd(AclPermissionType permissionType) {
+        String principalCmd = permissionType == ALLOW ? "--allow-principal" : "--deny-principal";
+        List<String> cmd = permissionType == ALLOW ? ALLOW_HOST_COMMAND : DENY_HOST_COMMAND;
+
+        List<String> fullCmd = new ArrayList<>();
+        for (KafkaPrincipal user : USERS) {
+            fullCmd.addAll(cmd);
+            fullCmd.addAll(Arrays.asList(principalCmd, user.toString()));
+        }
+
+        return fullCmd;
+    }
+
+    private void withAuthorizer(ClusterInstance cluster, Consumer<Authorizer> consumer) {
+        if (cluster.isKRaftTest()) {
+            List<Authorizer> allAuthorizers = new ArrayList<>();
+            allAuthorizers.addAll(asKraftCluster(cluster).brokers().values().stream()
```

Comment:
Do we really need `asKraftCluster` ? it seems to me `ClusterInstance#brokers` and  `ClusterInstance#controllers` should work well in this case.

Reply:
Done